internals/setup.js

```diff
@@ -0,0 +1,398 @@
+            )
+            return true
+          }
+        })
+      } else {
+        process.stdout.write(
+          'No remote added, run git remote add origin <url> to add one'
+        )
+        return false
+      }
+    })
+}
+
+/**
+ * Checks if we are under Git version control.
+ * If we are and this a clone of our repository the user is given a choice to
+ * either keep it or start with a new repository.
+ * @returns {Promise<boolean>}
+ */
+async function cleanCurrentRepository() {
+  const hasGitRepo = await hasGitRepository().catch(onError)
+
+  // We are not under Git version control. So, do nothing
+  if (hasGitRepo === false) {
+    return false
+  }
+
+  const isClone = await checkIfRepositoryIsAClone().catch(onError)
+
+  // Not our clone so do nothing
+  if (isClone === false) {
+    return false
+  }
+
+  const answer = await askUserIfWeShouldRemoveRepo()
+
+  if (answer === true) {
+    const spinner = ora('Removing current repository').start()
+    await removeGitRepository().catch(reason => {
+      spinner.fail(reason)
+      onError()
+    })
+    spinner.succeed('Repository removed')
+  }
+
+  return answer
+}
+
+/**
+ * Check Node.js version
+ * @param {!number} minimalNodeVersion
+ * @returns {Promise<any>}
+ */
+function checkNodeVersion(minimalNodeVersion) {
+  return new Promise((resolve, reject) => {
+    const spinner = ora('Checking node version').start()
+
+    exec('node --version', (err, stdout) => {
+      const nodeVersion = stdout.trim()
+      if (err) {
+        spinner.fail(`node version check failed\n${err}`)
+        reject()
+      } else if (compareVersions(nodeVersion, minimalNodeVersion) === -1) {
+        spinner.fail(
+          `You need Node.js v${minimalNodeVersion} or above but you have v${nodeVersion}`
+        )
+        reject()
+      }
+      spinner.succeed(`Node version ${nodeVersion} OK`)
+      resolve('Node version OK')
+    })
+  })
+}
+
+/**
+ * Check NPM version
+ * @param {!number} minimalNpmVersion
+ * @returns {Promise<any>}
+ */
+function checkNpmVersion(minimalNpmVersion) {
+  return new Promise((resolve, reject) => {
+    const spinner = ora('Checking npm version').start()
+    exec('npm --version', (err, stdout) => {
+      const npmVersion = stdout.trim()
+      if (err) {
+        spinner.fail(`npm version check failed\n${err}`)
+        reject()
+      } else if (compareVersions(npmVersion, minimalNpmVersion) === -1) {
+        spinner.fail(
+          `You need NPM v${minimalNpmVersion} or above but you have v${npmVersion}`
+        )
+        reject()
+      }
+      spinner.succeed(`npm version ${npmVersion} OK`)
+      resolve()
+    })
+  })
+}
+
+/**
+ * Install all packages
+ * @returns {Promise<any>}
+ */
+function installPackages() {
+  return new Promise((resolve, reject) => {
+    const spinner = ora(
+      'Installing dependencies... (This might take a while)'
```

Comment:
Is it possible to have the output of `npm install` here ?

Reply:
Dependency installation removed from script, we now install all dependencies in `presetup` which displays all output